### PR TITLE
arch: riscv: add an option for empty spurious interrupt handler

### PR DESCRIPTION
**Pull request by magp-nordic**
Add the possibility to disable fault handling in spurious interrupt handler on RISCs and replacce it with an infinite loop.

### CODE REVIEW

**Review by jaz1-nordic — APPROVED** [review]
Once you fix the spacing and trailing statements, LGTM.

File: arch/riscv/Kconfig [hunk 1]
```diff
@@ -225,6 +225,15 @@ config RISCV_RESERVED_IRQ_ISR_TABLES_OFFSET
 	  properly sized to take into account this offset. This is a hidden
 	  option which needs to be set per architecture and left alone.
 
+config RISCV_EMPTY_IRQ_SPURIOUS
+	bool "Create empty spurious interrupt handler"
+	default y if NRFE
+	default n
```
Comment (jaz1-nordic) [inline arch/riscv/Kconfig]: `bool` symbols implicitly default to n. Therefore, `default n` are (almost) always redundant.

**Comment by magp-nordic** [comment]
Compliance checks fail because this PR references symbol CONFIG_NRFE, that was added in https://github.com/masz-nordic/zephyr/pull/1 , but is not rebased on it.

File: arch/riscv/Kconfig [hunk 2]
```diff
@@ -225,6 +225,14 @@ config RISCV_RESERVED_IRQ_ISR_TABLES_OFFSET
 	  properly sized to take into account this offset. This is a hidden
 	  option which needs to be set per architecture and left alone.
 
+config RISCV_EMPTY_IRQ_SPURIOUS
+	bool "Create empty spurious interrupt handler"
+	default y if NRFE
```
Comment (masz-nordic) [inline arch/riscv/Kconfig]: This should be enabled by a config of specific application, not default for nrfe.

Comment (magp-nordic) [reply arch/riscv/Kconfig]: What if CONFIG_NRFE selected `RISCV_EMPTY_IRQ_SPURIOUS`? Would it be ok? Unless those symbols should not be connected.

Comment (masz-nordic) [reply arch/riscv/Kconfig]: Those should not be connected - `CONFIG_NRFE` is also enabled on APP side, which does not need this optimization.

Comment (magp-nordic) [reply arch/riscv/Kconfig]: Right, I keep confusing one core with another. Fixed.

**Review by masz-nordic — APPROVED** [review]

**Review by jaz1-nordic — APPROVED** [review]

**Pull request merged**